File: 2024/AdventOfCode/Tests/Day01Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day01Test {
    private const string Example = "Inputs/01-Example.txt";
    private const string Input = "Inputs/01.txt";
    private static Day01 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day01(Example);
            const string expected = "11";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day01(Input);
            const string expected = "1151792";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestExample() {
            _day = new Day01(Example);
            const string expected = "31";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day01(Input);
            const string expected = "21790168";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}

File: 2025/test/advent_of_code/day_01_test.exs
defmodule AdventOfCode.Day01Test do
  use ExUnit.Case

  import AdventOfCode.Day01

  @day 1
  @year 2025

  @example String.trim("""
           L68
           L30
           R48
           L5
           R60
           L55
           L1
           L99
           R14
           L82
           """)

  test "part1 example" do
    result = part1(@example)

    assert result == 3
  end

  test "part1" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part1(input)

    assert result == 1120
  end

  test "part2 example" do
    result = part2(@example)

    assert result == 6
  end

  test "part2" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part2(input)

    assert result == 6554
  end
end
